contracts/library/PoolLib.sol

```diff
@@ -238,24 +238,42 @@ library PoolLib {
     }
 
     /**
-        @dev Performing some checks before doing actual transfers.
+        @dev Performs all necessary checks for a `transfer` call.
     */
     function prepareTransfer(
         mapping(address => uint256) storage withdrawCooldown,
         mapping(address => uint256) storage depositDate,
-        address from,
+        uint256 totalCustodyAllowance,
+        uint256 fromBalance,
         address to,
         uint256 wad,
         IMapleGlobals globals,
         uint256 toBalance,
         uint256 recognizableLosses
     ) external {
-        // If transferring in or out of yield farming contract, do not update depositDate or cooldown
-        if (globals.isExemptFromTransferRestriction(from) || globals.isExemptFromTransferRestriction(to)) return;
+        require(fromBalance.sub(wad) >= totalCustodyAllowance,   "P:INSUFFICENT_TRANSFERABLE_BAL");  // User can only transfer tokens that aren't custodied
+        require(isReceiveAllowed(withdrawCooldown[to], globals), "P:RECIPIENT_NOT_ALLOWED");         // Recipient must not be currently withdrawing
+        require(recognizableLosses == uint256(0),                "P:RECOG_LOSSES");                  // If an LP has unrecognized losses, they must recognize losses through withdraw
+        updateDepositDate(depositDate, toBalance, wad, to);                                          // Update deposit date of recipient
+    }
+
+    /**
+        @dev Performs all necessary checks for a `transferByCustodian` call.
+        @dev From and to must always be equal. (TODO: Should we do this?)
+     */
+    function transferByCustodianChecks(address from, address to, uint256 amount, uint256 custodyAllowance) external {
+        require(to == from,                 "P:INVALID_RECEIVER");
```

Comment:
Should we consider implementing a version of `transferByCustodian` where `to != from`?

We're not ever going to be transferring tokens away in `MplRewards` but there may be an integration down the road where this is the case.